src/main/java/com/cleanroommc/groovyscript/compat/mods/essentialcraft/MagmaticSmeltery.java

```diff
@@ -0,0 +1,129 @@
+package com.cleanroommc.groovyscript.compat.mods.essentialcraft;
+
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.ModSupport;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.recipe.IRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import essentialcraft.api.OreSmeltingRecipe;
+import net.minecraftforge.oredict.OreDictionary;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collection;
+
+@RegistryDescription(reloadability = RegistryDescription.Reloadability.DISABLED, admonition={
+    @Admonition(value = "groovyscript.wiki.essentialcraft.magmatic_smeltery.note0", type = Admonition.Type.DANGER),
+    @Admonition(value = "groovyscript.wiki.essentialcraft.magmatic_smeltery.note1", type = Admonition.Type.WARNING),
+})
+public class MagmaticSmeltery extends VirtualizedRegistry<OreSmeltingRecipe> {
+    static boolean didReload = false;
+
+    @RecipeBuilderDescription(example = @Example(".input('blockIron').output('ingotGold').factor(3).color(0x0000ff)"))
+    public MagmaticSmeltery.RecipeBuilder recipeBuilder() {
+        return new MagmaticSmeltery.RecipeBuilder();
+    }
+
+    @Override
+    public void onReload() {
+        Collection<OreSmeltingRecipe> scripted = removeScripted();

```

Comment:
to fix the crash after reloading, do the following:
in `LateMixin`, add `essentialcraft` to the `modMixins` list.
create a mixin file called `mixin.groovyscript.essentialcraft.json` and make sure it points towards `mixin.essentialcraft`.
create the following file in the `mixins/essentialcraft` folder:
```java
package com.cleanroommc.groovyscript.core.mixin.essentialcraft;

import com.llamalad7.mixinextras.injector.wrapoperation.Operation;
import com.llamalad7.mixinextras.injector.wrapoperation.WrapOperation;
import essentialcraft.api.OreSmeltingRecipe;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;

@Mixin(value = OreSmeltingRecipe.class, remap = false)
public abstract class OreSmeltingRecipeMixin {

    @WrapOperation(method = "getColorFromItemStack", at = @At(value = "FIELD", target = "Lessentialcraft/api/OreSmeltingRecipe;color:I"))
    private static int getColorFromItemStack(OreSmeltingRecipe recipe, Operation<Integer> original) {
        if (recipe == null) return 16777215;
        return original.call(recipe);
    }

    @WrapOperation(method = "getLocalizedOreName", at = @At(value = "FIELD", target = "Lessentialcraft/api/OreSmeltingRecipe;oreName:Ljava/lang/String;"))
    private static String getLocalizedOreName(OreSmeltingRecipe recipe, Operation<String> original) {
        if (recipe == null) return "";
        return original.call(recipe);
    }

}
```

Reply:
Holy shit you actually fixed it. Thank you so much